lib/src/protocol/proxy_protocol/expect.rs

```diff
@@ -49,7 +49,7 @@ impl<Front: SocketHandler> ExpectProxyProtocol<Front> {
             container_frontend_timeout,
             frontend_buffer: [0; 232],
             frontend_readiness: Readiness {
-                interest: Ready::READABLE,
+                interest: Ready::READABLE | Ready::HUP | Ready::ERROR,
                 event: Ready::EMPTY,
```

Comment:
Seems very okay to me. Could you add a comment on the `new()` method that would say something like:

```rust
/// instantiate a new (state | whatever it is) with such and such readiness
```